test/dialect/postgresql/test_compiler.py

```diff
@@ -678,6 +678,177 @@ def test_create_index_with_ops(self):
             dialect=postgresql.dialect(),
         )
 
+    def test_nulls_not_distinct(self):
```

Comment:
it would be best if these tests broke out into individual cases using `@testing.combinations()`, this is a SQLAlchemy test suite wrapper around [pytest.mark.parametrize](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html):

like this:

```py
@testing.combinations(
   # each case is a 2-tuple of (lambda tbl: DDL, "expected ddl")
    (
         lambda tbl: CreateIndex(Index("test_idx1", tbl.c.data, unique=True,
            postgresql_nulls_not_distinct=True
         )),
         "CREATE UNIQUE INDEX test_idx1 ON test_tbl "
        "(data) NULLS NOT DISTINCT",        
    ),
    (
         lambda tbl: AddConstraint(schema.UniqueConstraint(
            tbl.c.data, name="uq_data1", postgresql_nulls_not_distinct=True,
         )),
           "ALTER TABLE test_tbl ADD CONSTRAINT uq_data1 UNIQUE "
            "NULLS NOT DISTINCT (data)",
    ),

    (
           # next case ...
    ),

    (
           # next case
    ),
   
    # etc ...

)
def test_nulls_not_distinct(self, expr_fn, expected):
    # same setup code

    tbl = Table(.... # same as before )

    # call the lambda using testing.resolve_lambda
    expr = testing.resolve_lambda(expr_fn, tbl=tbl)

   self.assert_compile(expr, expected, dialect=dd)

```

Reply:
Thank you. I've done this